pkg/storage/unified/resource/resource.proto

```diff
@@ -455,6 +465,8 @@ service ResourceIndex {
 
   // Used for efficient provisioning
   rpc Origin(OriginRequest) returns (OriginResponse);
+
+  rpc Index(IndexRequest) returns (IndexResponse);
```

Comment:
Is the intention to send in a document with this?  or to read out the results?

Reply:
i can create an Indexable interface instead and check if the resource server implements the Index method...if we don't want to expose this via grpc.  if the index/shard is somehow hosed/corrupt maybe we want the ability to let a client reload it?

Reply:
What if we remove `Index()` from the proto and initialize the index [here](https://github.com/grafana/grafana/blob/52c98de986bc913da40f66449eb9a44d4d16f0b9/pkg/storage/unified/resource/server.go#L168) when the resource server gets initialized?

Reply:
the problem with that is that Init method doesn't happen at startup.  it happens when the first call is made to access a resource.  so if the first call is search it would be delayed waiting for the index to build.  right?

Reply:
if we want to remove it from the proto we can just cast the interface to the `Indexer` interface and add an Index method there

Reply:
removed Index from the proto